python/dgl/contrib/cugraph/cugraph_storage.py

```diff
@@ -0,0 +1,254 @@
+        output_device=None,
+    ):
+        """
+        Return a DGLGraph which is a subgraph induced by sampling neighboring
+        edges ofthe given nodes.
+        See ``dgl.sampling.sample_neighbors`` for detailed semantics.
+        Parameters
+        ----------
+        seed_nodes : Tensor or dict[str, Tensor]
+            Node IDs to sample neighbors from.
+            This argument can take a single ID tensor or a dictionary of node
+            types and ID tensors. If a single tensor is given, the graph must
+            only have one type of nodes.
+        fanout : int or dict[etype, int]
+            The number of edges to be sampled for each node on each edge type.
+            This argument can take a single int or a dictionary of edge types
+            and ints. If a single int is given, DGL will sample this number of
+            edges for each node for every edge type.
+            If -1 is given for a single edge type, all the neighboring edges
+            with that edge type will be selected.
+        prob : str, optional
+            Feature name used as the (unnormalized) probabilities associated
+            with each neighboring edge of a node.  The feature must have only
+            one element for each edge.
+            The features must be non-negative floats, and the sum of the
+            features of inbound/outbound edges for every node must be positive
+            (though they don't have to sum up to one).  Otherwise, the result
+            will be undefined. If :attr:`prob` is not None, GPU sampling is
+            not supported.
+        exclude_edges: tensor or dict
+            Edge IDs to exclude during sampling neighbors for the seed nodes.
+            This argument can take a single ID tensor or a dictionary of edge
+            types and ID tensors. If a single tensor is given, the graph must
+            only have one type of nodes.
+        replace : bool, optional
+            If True, sample with replacement.
+        output_device : Framework-specific device context object, optional
+            The output device.  Default is the same as the input graph.
+        Returns
+        -------
+        DGLGraph
+            A sampled subgraph with the same nodes as the original graph, but
+            only the sampled neighboring edges.  The induced edge IDs will be
+            in ``edata[dgl.EID]``.
+        """
+        # change the seed_nodes from pytorch tensor to cudf series
+        if torch.is_tensor(seed_nodes):
+            seed_nodes = to_dlpack(seed_nodes)
+            seed_nodes = cudf.from_dlpack(seed_nodes)
+
+        parents_nodes, children_nodes = self.graphstore.sample_neighbors(
+            seed_nodes, fanout, edge_dir="in", prob=None, replace=False
+        )
+        # FIXME: why are children_nodes src and parents_nodes dst here
+        sample_df = cudf.DataFrame({"_SRC_": children_nodes, "_DST_": parents_nodes})
+        del parents_nodes, children_nodes
```

Comment:
Why do we need to delete them explicitly?  I think the peak memory consumption will be the same regardless of whether you delete them.